### ISSUE TITLE
🏎 Minor feature request: Support for Scale/Slider Response Types

### ISSUE THREAD

**Issue by mjoerke** [StanfordBDHG/ResearchKitOnFHIR#62]
### Problem

A common survey response type is one in which the participant response belongs to an ordered set of values, e.g.
* _What best describes your level of stress right now on a scale of 0-10 from not at all stressed \"0\" to Extremely stressed \"10\"?_ Here, the answer is an integer between 0 and 10.
* _During today at work, how would you rate the passage of time?_ Here, the answer belongs to a 7-point Likert scale: ["Very slow", "Moderately slow", "Slightly Slow", "Neither slow nor fast", "Slightly fast", "Moderately fast", "Very fast"]

ResearchKit supports scale responses with the [ORKScaleAnswerFormat](http://researchkit.org/docs/Classes/ORKScaleAnswerFormat.html) (for integers) and the [ORKTextScaleAnswerFormat](http://researchkit.org/docs/Classes/ORKTextScaleAnswerFormat.html), both of a which display a slider in the user interface.

Currently, researchers have to encode such questionnaires as a choice FHIR QuestionnaireItemType, which is converted to a ORKTextChoice ResearchKit type. For example, this would display the options 0-10 as a list of questionnaire options rather than a slider. 

### Solution

Provide support for choice questionnaire items that are on an ordered scale to be mapped to the appropriate ResearchKit type (either `ORKScaleAnswerFormat` or `ORKTextScaleAnswerFormat`). I am unsure if FHIR differentiates between choice types and _ordered_ choice types, but adding the option to display a slider could be achieved with metadata fields on a ValueSet or on the survey item itself. 

### Additional context

Here is an example of a survey item we are currently using in our study:

```
{
    "linkId": "70133a34-8597-4cdf-ec4a-6fb0f8ede585",
    "type": "choice",
    "text": "Please choose the value that best describes how you felt on average today about your workplace on a scale of 1-7, from I do not fit in my workplace \"0\" to I definitely fit in my workplace \"7\".",
    "required": false,
    "answerOption":
    [
        {
            "valueCoding":
            {
                "id": "6f5be914-eee6-458a-805e-a5cd84fac028",
                "code": "1",
                "system": "urn:uuid:8ac26004-8e67-4916-8f0a-0d70a70626e2",
                "display": "1"
            }
        },
        {
            "valueCoding":
            {
                "id": "8004d543-ea5d-4824-8832-9ac81c9c6235",
                "code": "2",
                "system": "urn:uuid:8ac26004-8e67-4916-8f0a-0d70a70626e2",
                "display": "2"
            }
        },
        {
            "valueCoding":
            {
                "id": "2b81b474-9c02-438e-9f22-13be890dc230",
                "code": "3",
                "system": "urn:uuid:8ac26004-8e67-4916-8f0a-0d70a70626e2",
                "display": "3"
            }
        },
        {
            "valueCoding":
            {
                "id": "40845d67-f5b8-49d8-8aac-65913160e7b1",
                "code": "4",
                "system": "urn:uuid:8ac26004-8e67-4916-8f0a-0d70a70626e2",
                "display": "4"
            }
        },
        {
            "valueCoding":
            {
                "id": "6dc6c684-10f1-48c6-9565-647aa2e3fc9e",
                "code": "5",
                "system": "urn:uuid:8ac26004-8e67-4916-8f0a-0d70a70626e2",
                "display": "5"
            }
        },
        {
            "valueCoding":
            {
                "id": "c03a4eda-801a-448c-8ff4-d83f655ba218",
                "code": "6",
                "system": "urn:uuid:8ac26004-8e67-4916-8f0a-0d70a70626e2",
                "display": "6"
            }
        },
        {
            "valueCoding":
            {
                "id": "ee1e918e-1dd5-4133-8f0c-a6cce0303cfd",
                "code": "7",
                "system": "urn:uuid:8ac26004-8e67-4916-8f0a-0d70a70626e2",
                "display": "7"
            }
        }
    ]
}
```

This currently displays as below
![IMG_4109](https://github.com/StanfordBDHG/ResearchKitOnFHIR/assets/16714168/53044f3b-b244-4653-bc22-eea606ad857f)

It would be nice if this answer could display as a slider

### Code of Conduct

- [X] I agree to follow this project's Code of Conduct and Contributing Guidelines

**Comment by vishnuravi** [comment]
Hi @mjoerke, this is a great idea. We can do this by implementing the [itemControl](http://hl7.org/fhir/R4/extension-questionnaire-itemcontrol.html) extension which can designate a [slider](http://hl7.org/fhir/R4/valueset-questionnaire-item-control.html#expansion). 

This should also be implemented in [Phoenix](https://github.com/StanfordBDHG/Phoenix) and I will create an other issue there.

**Comment by vishnuravi** [comment]
Hi @mjoerke, the slider feature is being implemented in PR #63. Please take a look and provide any feedback you have. Thanks!

**Comment by mjoerke** [comment]
@vishnuravi I checked out the PR and it looks great, thank you! Works exactly as expected and is easy to setup in the JSON format. We'll build this into our app once it's merged 👍
